Interface 2.0 LimeLight

andrewxsimon opened the pull request.

andrewxsimon committed:
Camera is now moved to the front

andrewxsimon committed:
Created CameraFSM and ShuffleboardFSM

andrewxsimon committed:
CV is using limelight, so no CV camera needed

andrewxsimon committed:
Deleted Shuffleboard FSM

andrewxsimon committed:
New Auto Mode selector

andrewxsimon committed:
Fixed build issues

andrewxsimon committed:
Whitespaces converted to tabs for build

andrewxsimon committed:
Javadoc comments for AutoSelector

andrewxsimon committed:
Build issues resolved

andrewxsimon committed:
Private variables organized

andrewxsimon committed:
Merge branch 'Interface-2.0' into Interface-2.0

andrewxsimon committed:
Get rid of unnecessary variable

andrewxsimon committed:
Cleaned up dead code

andrewxsimon committed:
Shuffleboard adds gyro and joystick data

andrewxsimon committed:
Shuffleboard adds gyro and joystick data

andrewxsimon committed:
Shuffleboard adds gyro and joystick data

andrewxsimon committed:
Merge branch 'Interface-2.0' of https://github.com/Tino-FRC-2473/FRC2022 into Interface-2.0

andrewxsimon committed:
Rebased main and added left out code

andrewxsimon committed:
Merge branch 'Interface-2.0' into Interface-2.0

andrewxsimon committed:
Fix values for joystick values on Shuffleboard

andrewxsimon committed:
Fix values for joystick values on Shuffleboard

andrewxsimon committed:
Fix values for joystick values on Shuffleboard

andrewxsimon committed:
LimeLight class created

andrewxsimon committed:
LimeLight class created

andrewxsimon committed:
LimeLight class created

andrewxsimon committed:
Limelight instance variable changed

andrewxsimon committed:
Limelight instance variable changed

andrewxsimon committed:
Limelight instance variable changed

andrewxsimon committed:
Build issues fixed

andrewxsimon committed:
Build issues fixed

andrewxsimon committed:
Build issues fixed

andrewxsimon committed:
CameraFSM not needed, replaced by LimeLight

andrewxsimon committed:
CameraFSM not needed, replaced by LimeLight

andrewxsimon committed:
CameraFSM not needed, replaced by LimeLight

andrewxsimon committed:
CameraFSM not needed, replaced by LimeLight

andrewxsimon committed:
CameraFSM not needed, replaced by LimeLight

andrewxsimon committed:
AutoSelector with getter method and new auto names

andrewxsimon committed:
AutoSelector with getter method and new auto names

andrewxsimon committed:
AutoSelector with getter method and new auto names

andrewxsimon committed:
Temporary auto names, will change

andrewxsimon committed:
Temporary auto names, will change

andrewxsimon committed:
Temporary auto names, will change

andrewxsimon committed:
Javadoc comment for LimeLight update()

andrewxsimon committed:
Javadoc comment for LimeLight update()

andrewxsimon committed:
Javadoc comment for LimeLight update()

andrewxsimon committed:
Javadoc comment for LimeLight update()

andrewxsimon committed:
Javadoc comment for LimeLight update()

andrewxsimon committed:
Merge branch 'Interface-2.0' of https://github.com/Tino-FRC-2473/FRC2022 into Interface-2.0

AG6GR commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -56,9 +55,11 @@ public void robotInit() {
 		grabberSystem = new GrabberFSM();
-		cameraSystem = new CameraFSM();
+
+		// new PowerDistribution(1, ModuleType.kRev).setSwitchableChannel(true);
```
Please cleanup unused code

andrewxsimon replied:
Cleaned up

andrewxsimon committed:
Commented out code deleted

andrewxsimon committed:
Commented out code deleted

andrewxsimon committed:
Commented out code deleted

RahulBansal1112 committed:
Drive (#12)

Improve auto paths, add additional teleop buttons to turn to specific targets, and integrate drive code with other subteams.

Co-authored-by: MGulati <mayandg@icloud.com>
Co-authored-by: Ryan Kwong <ryankwong478@gmail.com>
Co-authored-by: mukundkk <mukund.krishnakumar@gmail.com>
Co-authored-by: AG6GR <slhe@alumni.princeton.edu>
Co-authored-by: Siddharth Kadari <skadari270@student.fuhsd.org>

andrewxsimon committed:
Duplicate LimeLight update() method deleted

andrewxsimon committed:
Duplicate LimeLight update() method deleted

andrewxsimon committed:
Merge branch 'Interface-2.0' of https://github.com/Tino-FRC-2473/FRC2022 into Interface-2.0

andrewxsimon committed:
Merge branch 'Interface-2.0' of https://github.com/Tino-FRC-2473/FRC2022 into Interface-2.0

AG6GR commented on src/main/java/frc/robot/AutoSelector.java:
```diff
@@ -51,6 +51,14 @@ public void reset() {
+	 * @return name of current auto
+	 */
+	public String getSelectedAuto() {
+		return modeChooser.getSelected().name();
```
Actually, do we even need mDesiredMode? Seems the only reason is to repeat it back to SmartDashboard, but we're getting the values from SmartDashboard in the first place right?

andrewxsimon committed:
AutoSelector mDesiredMode removed
